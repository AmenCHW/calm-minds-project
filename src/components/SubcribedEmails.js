import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { collection, addDoc } from 'firebase/firestore';
import { ReactComponent as SendIcon } from '../icons/send.svg';
import { db } from '../firebase-config';

function SubcribedEmails() {
  const [newEmailInput, setNewEmailInput] = useState({});

  const navigate = useNavigate();

  // __________this handling subscription email input___________________
  const handleOnChange = (event) => {
    const {
      target: { name: keyName, value },
    } = event;
    setNewEmailInput((prev) => {
      return { ...prev, [keyName]: value };
    });
  };
  const handleSubmit = async (event) => {
    event.preventDefault();
    await addDoc(collection(db, 'subscribed-emails'), {
      ...newEmailInput,
    });
    // Clear the form
    setNewEmailInput({
      email: '',
    });

    navigate('/subscribe-thanks')
  };

  return (
    <div className="mx-auto flex justify-start lg:max-w-7xl">
      <form className="flex" onSubmit={handleSubmit}>
        <input
          className="border-2 border-[#718096] rounded-l-md w-auto placeholder:pl-2 pl-2"
          placeholder="Enter your e-mail"
          type="email"
          name="email"
          value={newEmailInput.email}
          onChange={handleOnChange}
        />
        <button
          className="p-2 border-y-2 border-r-2 border-[#718096] rounded-r-md bg-cyan-500 active:pr-1 active:pl-1 active:bg-cyan-400"
          type="submit"
          aria-label="Send"
        >
          <SendIcon />
        </button>
      </form>
    </div>
  );
}

export default SubcribedEmails;
