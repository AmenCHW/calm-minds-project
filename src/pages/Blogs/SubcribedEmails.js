import React, {  useState } from 'react';
import { collection, addDoc } from 'firebase/firestore';
// import { ref, listAll, getDownloadURL } from 'firebase/storage';
import { ReactComponent as SendIcon } from '../../icons/send.svg';
import { db } from '../../firebase-config';

function SubcribedEmails() {
  const [newEmailInput, setNewEmailInput] = useState({});

  


  // __________this handling subscribtion email input___________________
  const handleOnChange = (event) => {
    const {
      target: { name: keyName, value },
    } = event;
    // console.log('handleOnChange:', keyName);
    setNewEmailInput((prev) => {
      return { ...prev, [keyName]: value };
    });
  };
  const handleSubmit = async (event) => {
    event.preventDefault();
    // console.log(newEmailInput);
    await addDoc(collection(db, 'subscribed-emails'), {
      ...newEmailInput, // { "": muslim@gmail.com } => { "email": "muslim@gmail.com" }  key value shouldnot be empty so in email input we have name=email
    });
    // Clear the form
    setNewEmailInput({
      email: '',
    });
  };

  return (
    <div className="mx-auto  px-10 py-10">
      <div className="mx-auto lg:max-w-7xl ">
        
        {/* _____________________subscribtion__________________________________________________________________________________________________- */}
        <div className="mt-4 flex mx-2 ">
          <form className="flex" onSubmit={handleSubmit}>
            <input
              className="p-2 border-2 lg:px-3 px-0 border-[#718096] rounded-l-md"
              placeholder="Enter your e-mail"
              type="email"
              name="email"
              value={newEmailInput.email}
              onChange={handleOnChange}
            />
            <button
              className="p-2 border-y-2 border-r-2 border-[#718096] rounded-r-md bg-cyan-500 active:pr-1 active:pl-1 active:bg-cyan-400"
              type="submit"
              aria-label="Send"
            >
              {' '}
              <SendIcon />
            </button>
          </form>
        </div>  
      </div>
    </div>
  );
}

export default SubcribedEmails;
