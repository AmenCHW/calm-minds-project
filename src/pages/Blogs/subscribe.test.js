import React from "react";
 import {fireEvent} from "@testing-library/react"
import  SubcribedEmails from "./SubcribedEmails";
import renderer from 'react-test-renderer' 
import { renderHook } from "@testing-library/react-hooks";





test("first snapshot testing",()=>{
    const component=renderer.create( < SubcribedEmails/>).toJSON()  
    expect( component).toMatchSnapshot();
})

const btn=()=>{fireEvent(
   
    getByRole(container,  "button",),
    new MouseEvent('click', {
      bubbles: true,
      cancelable: true,
    }),
    
  )
}
console.log(btn)


// 1st time
  describe("when rendered", () => {
    it("expect email to be empty", () => {
      const { result } = renderHook(
        () => <SubcribedEmails />);
      console.log(result)
      expect(result.current.newEmailInput.email).toEqual('');
    });
  });



 
// // 2nd time
// test('expect email to be empty', () => {
//     const { result } = renderHook(() => SubcribedEmails())
  
//     expect(result.current.newEmailInput).toBe("")
    
//   })
 