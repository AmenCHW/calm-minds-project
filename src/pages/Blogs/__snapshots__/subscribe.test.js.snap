// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot testing 1`] = `
<div
  className="mx-auto  px-10 py-10"
>
  <div
    className="mx-auto lg:max-w-7xl "
  >
    <div
      className=" flex mr-4 "
    >
      <form
        className="flex"
        onSubmit={[Function]}
      >
        <input
          className=" border-2 lg:px-3 px-0 border-[#718096] rounded-l-md"
          name="email"
          onChange={[Function]}
          placeholder="Enter your e-mail"
          type="email"
        />
        <button
          aria-label="Send"
          className="p-2 border-y-2 border-r-2 border-[#718096] rounded-r-md bg-cyan-500 active:pr-1 active:pl-1 active:bg-cyan-400"
          type="submit"
        >
           
          <svg>
            send.svg
          </svg>
        </button>
      </form>
    </div>
  </div>
</div>
`;
